src/Log/Engine/BaseLog.php

```diff
@@ -61,6 +68,18 @@ public function __construct(array $config = [])
         if (!empty($this->_config['types']) && empty($this->_config['levels'])) {
             $this->_config['levels'] = (array)$this->_config['types'];
         }
+
+        $formatterConfig = $this->_config['formatter'] ?? null;
+        if (isset($formatterConfig)) {
```

Comment:
Formatters are optional for custom loggers especially for backwards compatibility.